daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Hasher.scala

```diff
@@ -69,15 +67,15 @@ private[speedy] object Hasher {
               case SEnum(_, constructor) =>
                 loop(cmdsRest, constructor.hashCode :: stack)
               case SRecord(_, _, values) =>
-                loop(pushOrderedValues(values.iterator().asScala, cmdsRest), stack)
+                loop(pushOrderedValues(values.size, values.iterator().asScala, cmdsRest), stack)
               case SVariant(_, variant, value) =>
                 loop(Value(value) :: Mix(variant.hashCode) :: cmdsRest, stack)
               case SStruct(_, values) =>
-                loop(pushOrderedValues(values.iterator().asScala, cmdsRest), stack)
+                loop(pushOrderedValues(values.size, values.iterator().asScala, cmdsRest), stack)
               case SOptional(opt) =>
-                loop(pushOrderedValues(opt.iterator, cmdsRest), stack)
+                loop(pushOrderedValues(opt.fold(0)(_ => 1), opt.iterator, cmdsRest), stack)
```

Comment:
You can just use `opt.size` here.

Reply:
implicit conversion from Option to Traversable are considered warts.